File: android/src/net/sf/keyice/IAEF.java
package net.sf.keyice;

import java.util.Random;

public class IAEF {
	public static String enoutput,deoutput,toencrypt,todecrypt;
	public static boolean val;
	public static int key,password;

	public void simpleEncrypt(String toEncry )
	{
		int charNum = 0;
	    String text="";

	    for(int i=0;i<toEncry.length();i++)
	    {
	        charNum = toEncry.charAt(i);
	        charNum=charNum+1;
	        text+=(char)charNum;
	    }
	        enoutput=text;
	        
		
	
	}
	
	public void simpleDecrypt(String toDecry )
	{
		int charNum = 0;
	    String text="";

	    for(int i=0;i<toDecry.length();i++)
	    {
	        charNum = toDecry.charAt(i);
	        charNum=charNum-1;
	        text+=(char)charNum;
	    }
	        deoutput=text;
	        
		
	
	}
	public void keyGen ()
	{
		Random rand = new Random();
		int range = 999999 - 111111+ 1;
		key = ( rand.nextInt((range) + 1) );
		password = key % 10 + 1;
	}
	public void advancedEncrypt(String toEncry )
	{
		int charNum = 0;
	    String text="";

	    for(int i=0;i<toEncry.length();i++)
	    {
	        charNum = toEncry.charAt(i);
	        charNum=charNum+password;
	        text+=(char)charNum;
	    }
	        enoutput=text;
	        
		
	
	}
	public void advancedDecrypt(String toDecry ,int key )
	{
		int charNum = 0;
	    String text="";
	    password = key % 10 + 1;

	    for(int i=0;i<toDecry.length();i++)
	    {
	        charNum = toDecry.charAt(i);
	        charNum=charNum-password;
	        text+=(char)charNum;
	    }
	        deoutput=text;
	}
	public IAEF()
	{
		enoutput = "hai keyice";
		deoutput = "hai keyice";
		toencrypt ="";
		todecrypt = "";
		val = false;
	}
	
	

}

File: pc/iaef.cpp
#include "iaef.h"
#include <QMessageBox>

IAEF::IAEF()
{
}

void IAEF::simpleEncrypt(string toEncry, string& output)
{

    int charNum = 0;
    string text="";

    for(int i=0;i<toEncry.length();i++)
    {
        charNum = int(toEncry[i]);
        charNum=charNum+1;
        text+=char(charNum);
    }
        output=text;
}



void IAEF::simpleDecrypt(string toDecry, string &output)
{

    int charNum = 0;
     string text="";

    for(int i=0;i<toDecry.length();i++)
    {
        charNum = int(toDecry[i]);
        charNum=charNum-1;
        text+=char(charNum);
    }
        output=text;

}


void IAEF::keyGen(long int &key)
{
    srand(time(0));

    start:

    long int keyCode = rand();
    int random = keyCode % (1 - 10);
    if(random==0) goto start;
     key=keyCode;
}

void IAEF::keyEncrypt(string toEncry, long int key, string &encrypted)
{
    int keyCode = key % (1 - 10);

    int charNum = 0;
    string text = "";

      for(int i=0;i<toEncry.length();i++)
      {
          charNum = int(toEncry[i]);
          charNum=(charNum + keyCode);


          text+=char(charNum);
      }
      encrypted = text;

}


void IAEF::keyDecrypt(string toDecry, long int key, string &decrypted)
{


    int keyCode = key % (1 - 10);

    int charNum = 0;
    string text = "";

      for(int i=0;i<toDecry.length();i++)
      {
          charNum = int(toDecry[i]);
          charNum=(charNum - keyCode);


          text+=char(charNum);
      }
      decrypted = text;


}

